posthog/session_recordings/session_recording_helpers.py

```diff
@@ -149,16 +151,22 @@ def preprocess_replay_events(events: List[Event], max_size_bytes=1024 * 1024) ->
     3. If not, we split out the "full snapshots" from the rest (they are typically bigger) and send them individually, trying one more time to group the rest, otherwise sending them individually
     """
 
+    if isinstance(_events, Generator):
+        # we check the first item in the events below so need to be dealing with a list
+        events = list(_events)
+    else:
+        events = _events
```

Comment:
we sometimes pass a generator into this function, and then check `events[0]` but generators aren't subscriptable... so mypy was very unhappy with us